### PR TITLE
feat: add opcua asset type check

### PR DESCRIPTION
**Pull request by Elsie4ever**
Add opcua asset type check

level 0 output:
image

level 1 output:
image

level 2 output:
image

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

File: azext_edge/edge/providers/check/opcua.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+                if detail_level == ResourceOutputDetailLevel.verbose.value:
+                    # remove repeated labels
+                    non_repeated_labels = list(set(labels))
+
+                    if len(non_repeated_labels) > 0:
+                        check_manager.add_display(
+                            target_name=target_asset_types,
+                            namespace=namespace,
+                            display=Padding(
+                                "[yellow](Only non repeatative labels will be displayed)[/yellow]",
```
Comment (digimaun) [inline azext_edge/edge/providers/check/opcua.py:137]: Is this 'Only non-repetitive labels ...' (note spelling error) display needed?  The prior display can say "Detected X **unique** labels" to simplify.

Comment (Elsie4ever) [reply azext_edge/edge/providers/check/opcua.py:137]: yeah good idea! will update!

File: azext_edge/edge/providers/check/base.py [hunk 2]
```diff
@@ -522,7 +523,25 @@ def evaluate_pod_health(
     target_service_pod = f"pod/{pod}"
     check_manager.add_target_conditions(target_name=target, namespace=namespace, conditions=[f"{target_service_pod}.status.phase"])
     diagnostics_pods = get_namespaced_pods_by_prefix(prefix=pod, namespace=namespace, label_selector=service_label)
```
Comment (digimaun) [inline azext_edge/edge/providers/check/base.py:525]: diagnostic_pods? This is from mq, is this carry over from a prior refactor?

File: azext_edge/edge/providers/check/base.py [hunk 3]
```diff
@@ -690,3 +710,8 @@ def resources_grouped_by_namespace(resources: List[dict]):
 
 def filter_by_namespace(resources: List[dict], namespace: str) -> List[dict]:
     return [resource for resource in resources if get_resource_namespace(resource) == namespace]
+
+
+def generate_target_resource_name(api_info: EdgeResourceApi, resource_kind: str) -> str:
+    resource_plural = api_info._kinds[resource_kind] if api_info._kinds else f"{resource_kind}s"
```
Comment (digimaun) [inline azext_edge/edge/providers/check/base.py:716]: Right now, we have to do api_info.kinds or api_info.get_resources to populate api_info._kinds dictionary. We will improve.

File: azext_edge/edge/providers/check/common.py [hunk 4]
```diff
@@ -163,6 +163,9 @@ class DataprocessorDestinationStageType(ListableEnum):
 
 # Check constants
 ALL_NAMESPACES_TARGET = '_all_'
+# when there are runtime resources related to the service but not
+# related to any service resource, use this constant as the resource name
+CORE_SERVICE_RUNTIME_RESOURCE = "coreServiceRuntimeResources"
```
Comment (digimaun) [inline azext_edge/edge/providers/check/common.py:168]: Should we call this plural resource as in resources?

File: azext_edge/edge/providers/check/base.py (hunk 3 above)
Comment (digimaun) [reply azext_edge/edge/providers/check/base.py:716]: That prior mentioned pre-req is being called earlier in the flow right?

Comment (digimaun) [reply azext_edge/edge/providers/check/base.py:716]: no, thanks

File: azext_edge/edge/providers/check/common.py (hunk 4 above)
Comment (Elsie4ever) [reply azext_edge/edge/providers/check/common.py:168]: will update to coreServiceRuntimeResource!

File: azext_edge/edge/params.py [hunk 5]
```diff
@@ -93,7 +93,7 @@ def load_iotops_arguments(self, _):
         context.argument(
             "ops_service",
             options_list=["--ops-service", "--svc"],
-            choices=CaseInsensitiveList(["mq", "dataprocessor", "lnm"]),
+            choices=CaseInsensitiveList(["mq", "dataprocessor", "lnm", "opcua"]),
```
Comment (digimaun) [inline azext_edge/edge/params.py:96]: We should add the opcua broker resource_kinds below right ?

Comment (Elsie4ever) [reply azext_edge/edge/params.py:98]: updated thanks!

**Review by digimaun — APPROVED** [review]

**Pull request merged**